pkgs/development/tools/xcbuild/platform.nix

```diff
@@ -0,0 +1,578 @@
+{ stdenv, sdk, writeText, platformName, xcbuild }:
+
+let
+
+  Info = {
+    CFBundleIdentifier = platformName;
+    Type = "Platform";
+  };
+
+  Version = {
+    ProjectName = "OSXPlatformSupport";
+  };
+
+  PackageTypes = [
+    {   Type = "PackageType";
+        Identifier = "com.apple.package-type.mach-o-executable";
+        Name = "Mach-O Executable";
+        Description = "Mach-O executable";
+        DefaultBuildSettings = {
+            EXECUTABLE_PREFIX = "";
+            EXECUTABLE_SUFFIX = "";
+            EXECUTABLE_NAME = "$(EXECUTABLE_PREFIX)$(PRODUCT_NAME)$(EXECUTABLE_VARIANT_SUFFIX)$(EXECUTABLE_SUFFIX)";
+            EXECUTABLE_PATH = "$(EXECUTABLE_NAME)";
+        };
+        ProductReference = {
+            FileType = "compiled.mach-o.executable";
+            Name = "$(EXECUTABLE_NAME)";
+            IsLaunchable = "YES";
+        };
+    }
+    {   Type = "PackageType";
```

Comment:
Yeah, so all of these aren't included right now in xcbuild. But it might be added later.